src/engine.rs

```diff
@@ -164,17 +205,29 @@ impl Engine {
 
     pub fn with_resources<'a>(&'a mut self, resources: &[Resource]) -> &'a mut Engine {
         self.blocker.with_resources(resources);
+        self.cosmetic_cache.use_resources(resources);
         self
     }
 
     pub fn resource_add<'a>(&'a mut self, resource: Resource) -> &'a mut Engine {
-        self.blocker.resource_add(resource);
+        self.blocker.resource_add(&resource);
+        self.cosmetic_cache.add_resource(&resource);
```

Comment:
are we using the same instance of Engine in the end for network and cosmetic filtering? If not, this will unnecessarily increase memory use and should keep track of which one it is instead

Reply:
I reused the main (i.e. not the regional or custom) Engine instance in my `brave-core` PR.

Reply:
Are resources ever added to the other Engines?

Reply:
ok, if reusing that's fine, will reduce the overhead when combining all filters to one engine instance.

But would that mean cosmetic filters don't apply for regional?